src/CommonLib/utility.cxx

```diff
@@ -627,3 +627,30 @@ UTILITY::LogEvent(
         fwprintf(stderr, L"ERROR: %s\n", pstrMsg);
     }
 }
+
+VOID
+UTILITY::LogEventF(
+    _In_ HANDLE  hEventLog,
+    _In_ WORD    dwEventInfoType,
+    _In_ DWORD   dwEventId,
+    _In_ LPCWSTR pstrMsg,
+    ...
+)
+{
+    STACK_STRU ( strEventMsg, 256 );
+
+    va_list argsList;
+    va_start(argsList, pstrMsg);
```

Comment:
Comment saying "va_start needs to know the last argument of the function call is to get the arguments.

Reply:
I like this LogEventF. It avoids duplicated code.

Reply:
one optimization is to avoid calling strEventMsg.SafeSnwprintf (to avoid build stru and string copy) if less than 4 input parameters